packages/store/src/storage.ts

```diff
@@ -102,11 +103,13 @@ class Storage {
   public readonly logger: Logger;
   public readonly uplinks: ProxyInstanceList;
   private searchService: Search;
+  private deniedVersionFilter: DeniedVersionFilter;
   public constructor(config: Config, logger: Logger) {
     this.config = config;
     this.logger = logger.child({ module: 'storage' });
     this.uplinks = setupUpLinks(config, this.logger);
     this.searchService = new Search(config, this.logger);
+    this.deniedVersionFilter = new DeniedVersionFilter(config);
     this.filters = null;
```

Comment:
The idea is great, I just need to give some thoughts to this, because we might end up with two filter systems. There is already a plugin that meant to filter out versions.